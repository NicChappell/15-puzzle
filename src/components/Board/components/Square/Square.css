.square {
    align-items: center;
    background-color: cornflowerblue;
    border: 2px solid black;
    box-sizing: border-box;
    display: flex;
    height: 100%;
    justify-content: center;
    max-height: 25%;
    max-width: 25%;
    position: absolute;
    width: 100%;
}

.value {
    font-family: 'Delius Unicase', cursive;
    font-size: 32px;
    font-weight: 700;
}