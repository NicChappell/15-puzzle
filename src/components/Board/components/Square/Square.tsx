import { Tile } from "../../../../game";
import "./Square.css";

export interface ISquare {
  tile: Tile;
}

const Square = ({ tile }: ISquare) => {
  // destructure tile
  const { value, x, y } = tile;

  console.log("value: ", value);
  console.log("x: ", x);
  console.log("y: ", y);

  // set left property
  let left = "";
  switch (x) {
    case 1:
      left = "0%";
      break;
    case 2:
      left = "25%";
      break;
    case 3:
      left = "50%";
      break;
    case 4:
      left = "75%";
      break;
  }

  // set top property
  let top = "";
  switch (y) {
    case 1:
      top = "0%";
      break;
    case 2:
      top = "25%";
      break;
    case 3:
      top = "50%";
      break;
    case 4:
      top = "75%";
      break;
  }

  const handleClick = () => console.log(value);

  return (
    <div className="square" onClick={handleClick} style={{ left, top }}>
      <span className="value">{value}</span>
    </div>
  );
};

export default Square;
