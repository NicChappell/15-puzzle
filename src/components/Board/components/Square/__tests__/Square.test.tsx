import { render } from "@testing-library/react";
import Square from "../Square";

const startingProps = {
  tile: {
    value: 1,
    x: 1,
    y: 1,
  },
};

describe("Square", () => {
  it("renders", () => {
    const { container } = render(<Square {...startingProps} />);

    expect(container).toMatchSnapshot();
  });
});
