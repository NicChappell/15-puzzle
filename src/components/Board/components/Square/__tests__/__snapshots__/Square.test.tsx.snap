// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders 1`] = `
<div>
  <div
    class="game-board"
  >
    <p>
      Game Board
    </p>
  </div>
</div>
`;
